docs/develop/go/temporal-nexus.mdx

```diff
@@ -0,0 +1,476 @@
+
+### Install latest tcld CLI and generate certificates
+
+First install the latest version of the `tcld` CLI and generate some certificates for mTLS Worker authentication:
+
+```
+brew install temporalio/brew/tcld
+tcld gen ca --org temporal -d 1y --ca-cert ca.pem --ca-key ca.key
+```
+
+### Create caller and handler Namespaces
+
+Before deploying to Temporal Cloud, it's necessary to create the appropriate Namespaces for both the caller and handler.
+
+```
+tcld login
+
+tcld namespace create \
+	--namespace <your-caller-namespace> \
+	--region us-west-2 \
+	--ca-certificate-file 'path/to/your/ca.pem' \
+	--retention-days 1
+
+tcld namespace create \
+	--namespace <your-target-namespace> \
+	--region us-west-2 \
+	--ca-certificate-file 'path/to/your/ca.pem' \
+	--retention-days 1
+```
+
+Alternatively, you can create Namespace via the UI: [https://cloud.temporal.io/Namespaces](https://cloud.temporal.io/Namespaces)
+
+### Create a Nexus Endpoint to route requests from caller to handler
+
+To create a Nexus Endpoint you must have a `Developer` account role or higher, and have `NamespaceAdmin` permission on the `--target-namespace`.
+
+```
+tcld nexus endpoint create \
+  --name <myendpoint> \
+  --target-task-queue my-handler-task-queue \
+  --target-namespace <my-target-namespace.account> \
+  --allow-namespace <my-caller-namespace.account> \
+  --description-file description.md
+```
+
+The `--allow-namespace` is used to build an Endpoint allowlist of caller Namespaces that can use the Nexus Endpoint, as described in Runtime Access Control.
+
+Alternatively, you can create a Nexus Endpoint via the UI: [https://cloud.temporal.io/nexus](https://cloud.temporal.io/nexus)
+
+### Run Workers Connected to Temporal Cloud
+
+[View the source code](https://github.com/temporalio/samples-go/tree/main/nexus) in the context of the rest of the application code.
+
+Run the handler Worker:
+
+```go 
+cd handler
+
+go run ./worker \
+	-target-host <your-target-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-target-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+Run the caller Worker:
+
+```go 
+cd caller
+
+go run ./worker \
+	-target-host <your-caller-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-caller-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+### Start a caller Workflow
+
+In order to start the caller Workflow, run the starter.
+
+```go 
+cd caller
+
+go run ./starter \
+	-target-host <your-caller-namespace.account>.tmprl.cloud:7233 \
+	-namespace <your-caller-namespace.account> \
+	-client-cert 'path/to/your/ca.pem' \
+	-client-key 'path/to/your/ca.key'
+```
+
+This will result in:
+
+```
+2024/07/23 19:57:40 Workflow result: Nexus Echo 👋
+2024/07/23 19:57:40 Workflow result: ¡Hola! Nexus 👋
+```
+
+### Observability in Temporal Cloud
```

Comment:
Observability is not limited to Temporal Cloud. The only different between the self-hosted and Cloud version is that Cloud supports bidirectional links, and OSS still hasn't caught up due to release timing.

I'd try to clarify this a bit. Maybe put observability in the local section instead of here?

Reply:
Thanks for the explanation! I changed the titles to more general ones.